src/directory/mmap_directory.rs

```diff
@@ -215,20 +242,24 @@ impl MmapDirectory {
     /// Returns an error if the `directory_path` does not
     /// exist or if it is not a directory.
     pub fn open<P: AsRef<Path>>(directory_path: P) -> Result<MmapDirectory, OpenDirectoryError> {
-        Self::open_with_access_pattern_impl(directory_path.as_ref(), None)
+        Self::open_with_access_pattern_impl(directory_path.as_ref())
     }
 
+    #[cfg(unix)]
     /// Opens a MmapDirectory in a directory, with a given access pattern.
+    ///
+    /// This is only supported on unix platforms.
     pub fn open_with_madvice<P: AsRef<Path>>(
         directory_path: P,
         madvice: Advice,
     ) -> Result<MmapDirectory, OpenDirectoryError> {
-        Self::open_with_access_pattern_impl(directory_path.as_ref(), Some(madvice))
+        let dir = Self::open_with_access_pattern_impl(directory_path.as_ref())?;
+        dir.with_advice(Some(madvice));
+        Ok(dir)
     }
 
     fn open_with_access_pattern_impl(
```

Comment:
we can keep the helper function to avoid the monomorphization on `P`, but the name is nonsense now.